qa/smoke-test-http/src/javaRestTest/java/org/elasticsearch/http/snapshots/RestGetSnapshotsIT.java

```diff
@@ -322,6 +322,33 @@ public void testSortAfterStartTime() throws Exception {
         assertThat(allBeforeStartTimeDescending(startTime1 - 1), empty());
     }
 
+    public void testLargeChunkedResponses() throws Exception {
```

Comment:
This is a bit of a nasty test admittedly, but I really like having at least one integration test that actually forces this chunked response implementation to run end-to-end through the networking code and not just testing networking and chunked serialization in isolation each.

Reply:
Not sure that'll add much coverage that we don't already have in a more pinpointed way in `Netty4HttpPipeliningHandlerTests`?

Reply:
I think there is still value in this. For one, `encodeChunk` does not run there. Not saying that that integration will lead to issues (I see none), but some verification of pipelining with chunking end-to-end seems valuable.

Reply:
It's not easy to add this kind of test though and it'll again be unit-testy like `Netty4HttpPipeliningHandlerTests` because I can't really force pipelining via out of order messages directly in any sort of systematic fashion. Do you really think this is worth the effort? We never added any such testing for the existing pipelining and single rest requests directly, it's all just unit-tests pretty much here as well.